### PR TITLE
feat: qna recognizer

### PR DESCRIPTION
**Pull request by alanlong9278**
## Description

qna & custom recognizer
 - [x] New recognizer drop down : Default recognizer , Regex recognizer, Custom recognizer
 - [x] QnA in all up view (basic) : support editing in manaco-editor, list view and syntax diagnostic
 - [x] Refactor publisher and support QnA build
 - [x] start bot with publish page 
 - [x] ux for form editor
 - [x] Legacy bot migration to new recognizer hierarchy（Need to discuss, It's expensive to migration）
 - [x] Refactor lu & qna publish logic
 - [x] QnA in all up view (advanced) : advanced editing in list view
 - [x] Popup window for publishing
 - [x] Custom recognizer : trigger wizard for OnIntent and OnQnaMatch
 - [x] New trigger wizard : OnQnAMatch
 - [x] New trigger wizard : OnChooseIntent
 - [x] New trigger wizard : OnIntent(AdaptiveCard)
 - [x] Validation check and warning for recognizer switching

## Task Item
https://github.com/microsoft/BotFramework-Composer/issues/2394
#3463

### CODE REVIEW

**Comment by hibrenda** [comment]
issues may need to fix:
Legacy bot migration 
- legacy bot should be open and matched into luis+qna (if there is no .qna or .lu file exists)

OnQnAMatch trigger and Inline-QnA editing:
- resizing the inline editor (need bigger editor for editing)
- may add default action for OnQnAMatch trigger

Drop down list switching
- Switch from regex into luis+qna, regex intent should be convert into luis hoisting 
- Switch from luis+qna to regex, if onQnAMatch trigger exist, it will pop up warning

Need more discussion for customized recognizer/trigger wizard & value recognizer

**Comment by hibrenda** [comment]
more comments from today's discussion:
- no mater what recognizer type is defined , trigger wizard drop down list should be consist (eg. OnQnAMatch will be showed)
- switching from default to regex need to pop up warning if OnQnAMatch or OnChooseIntent exist
   add OnQnAMatch will pop up warning as well if it is regex recognizer
- IntentRecognized trigger only hook with intent name if it is custom recognizer. No inline hook up for OnIntent and OnQnAMatch trigger either from trigger wizard or property view.

**Comment by hibrenda** [comment]
Attached the figma design https://www.figma.com/file/LlTlh5vXwq91zjGnFtrUUR6l/Composer-master-design-spec-%2F-UI-library?node-id=2392%3A20262

**Comment by hibrenda** [comment]
[Design Spec] https://microsoft.sharepoint.com/:w:/t/ConversationalAI785/EWSmZ6UhsMlKnpbc1lKhhbEB0GZdoLCAtMf8rlHdHDZHvg?e=AXzdQq

File: Composer/packages/client/src/pages/qna/code-editor.tsx [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+import { CodeEditorSettings } from '@bfc/shared';
+
+import { StoreContext } from '../../store';
+
+interface CodeEditorProps extends RouteComponentProps<{}> {
+  dialogId: string;
+}
+
+const lspServerPath = '/lu-language-server';
+const CodeEditor: React.FC<CodeEditorProps> = (props) => {
```
Comment (yeze322) [inline Composer/packages/client/src/pages/qna/code-editor.tsx:21]: CodeEditor is too generic, can it be QnAEditor?

File: Composer/packages/client/src/pages/qna/index.tsx [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+import { Page } from '../../components/Page/Page';
+
+import TableView from './table-view';
+const CodeEditorWrapper = React.lazy(() => import('./codeEditorWrapper'));
+
+interface LGPageProps extends RouteComponentProps<{}> {
+  dialogId?: string;
+}
+
+const LGPage: React.FC<LGPageProps> = (props) => {
```
Comment (yeze322) [inline Composer/packages/client/src/pages/qna/index.tsx:26]: Naming - it's QnAPage instead of LGPage?

Comment (alanlong9278) [reply Composer/packages/client/src/pages/qna/index.tsx:26]: @liweitian  Can u see it?

File: Composer/packages/client/src/pages/qna/code-editor.tsx (hunk 1 above)
Comment (alanlong9278) [reply Composer/packages/client/src/pages/qna/code-editor.tsx:21]: @liweitian Can u see it?

**Comment by boydc2014** [comment]
close in favor of  https://github.com/microsoft/BotFramework-Composer/pull/3704

**Pull request closed without merging**